### ISSUE TITLE
Transformer.generate_square_subsequent_mask

### ISSUE THREAD

**Issue by roderickObrist** [pytorch/pytorch#28272]
## 🐛 Bug

generate_square_subsequent_mask spits out reverse masks across pytorch versions 1.2 and 1.3.

Fortunately the replication code is quite simple

```
import torch
print(torch.__version__)
print(torch.nn.Transformer.generate_square_subsequent_mask(None, 4))


1.2.0
tensor([[0., -inf, -inf, -inf],
        [0., 0., -inf, -inf],
        [0., 0., 0., -inf],
        [0., 0., 0., 0.]])

1.3.0a0+80ae6d2
tensor([[0., 0., 0., 0.],
        [-inf, 0., 0., 0.],
        [-inf, -inf, 0., 0.],
        [-inf, -inf, -inf, 0.]])



cc @ezyang @gchanan @zou3519

**Comment by phongnhhn92** [comment]
I am about to make the same issue.

**Comment by rudvlf0413** [comment]
I have a same issue...

**Comment by soumith** [comment]
cc: @zhangguanheng66

**Comment by zhangguanheng66** [comment]
@soumith It's issue related to type promotion.
The master branch has been fixed by https://github.com/pytorch/pytorch/pull/28253

**Comment by phongnhhn92** [comment]
Hello, does this new build available on pytorch conda package ?

**Comment by zhangguanheng66** [comment]
Not yet. We may need a point release for that.

**Comment by phongnhhn92** [comment]
So if i uninstall the conda pytorch and install it again using pip then it might be fixed ?

**Comment by soumith** [comment]
no
